src/main/java/Duke.java

```diff
@@ -1,10 +1,177 @@
+        if (isTaskUnmarked) {
+            Task currentTask = this.tasklist.getTask(position);
+            return horizontalBorder + "OK, I've marked this task as not done yet:\n" + currentTask + "\n" + horizontalBorder;
+        } else if (this.tasklist.getCount() == 0) {
+            throw new DukeException(INVALID_ACCESS_EMPTY_TASKLIST);
+        } else {
+            throw new DukeException(taskNotFoundMessage());
+        }
+    }
+
+    private String deleteTaskMessage(int position) throws DukeException {
+        try {
+            Task deletedTask = this.tasklist.deleteTaskAtPos(position);
+            return horizontalBorder + "Noted. I've removed this task:\n" + deletedTask + "\n" + "" + "Now you have " + this.tasklist.getCount() + " tasks in the list.\n" + horizontalBorder;
+        } catch (IndexOutOfBoundsException e){
+            if (this.tasklist.getCount() == 0){
+                throw new DukeException(INVALID_ACCESS_EMPTY_TASKLIST);
+            } else {
+                throw new DukeException(taskNotFoundMessage());
+            }
+        }
+    }
+
+    private boolean isInteger(String value) {
+        try {
+            Integer.parseInt(value);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    private String makeToDoFromInput(String input) throws DukeException {
+        String description = input.substring("todo".length()).strip();
+        if (!description.equals("")) {
+            ToDo newToDo = new ToDo(description);
+            this.tasklist.add(newToDo);
+            return addTaskMessage(newToDo.toString());
+        } else {
+            throw new DukeException(INVALID_TODO_INPUT);
+        }
+    }
+
+    private String makeDeadlineFromInput(String input) throws DukeException {
+        String[] stringArray = input.substring("deadline".length()).strip().split("/by");
+        if (stringArray.length > 1) {
+            Deadline newDeadline = new Deadline(stringArray[0].strip(), stringArray[1].strip());
+            this.tasklist.add(newDeadline);
+            return addTaskMessage(newDeadline.toString());
+        } else {
+            throw new DukeException(INVALID_DEADLINE_INPUT);
+        }
+    }
+
+    private String makeEventFromInput(String input) throws DukeException{
+        String[] stringArray = input.substring("event".length()).strip().split("/at");
+        if (stringArray.length > 1) {
+            Event newEvent = new Event(stringArray[0].strip(), stringArray[1].strip());
+            this.tasklist.add(newEvent);
+            return addTaskMessage(newEvent.toString());
+        } else {
+            throw new DukeException(INVALID_EVENT_INPUT);
+        }
+    }
+
+    public void run(){
+        System.out.println(welcomeMessage());
+        Scanner scan = new Scanner(System.in);
+        String s = scan.nextLine();
+        boolean exitNow = false;
```

Comment:
You could consider using a variable name which sounds more like a Boolean such as isExit.